fs_gcsfs/_gcsfs.py

```diff
@@ -66,8 +71,20 @@ def __init__(self,
         if self.client is None:
             self.client = Client()
 
-        self.bucket = self.client.get_bucket(self._bucket_name)
-        super(GCSFS, self).__init__()
+        try:
+            self.bucket = self.client.get_bucket(self._bucket_name)
+        except google.api_core.exceptions.NotFound as err:
+            raise CreateFailed("The provided bucket does not seem to exist") from err
+        except google.api_core.exceptions.Forbidden as err:
+            raise CreateFailed("You don't have access to the provided bucket") from err
+
+        if create:
+            root_marker = self._get_blob(forcedir(root_path))
+            if not root_marker:
+                blob = self.bucket.blob(forcedir(root_path))
```

Comment:
I don't remember if that's possible but maybe `self.makedir(root_path)` could be used instead?

Reply:
this was my first take too, but `makedir` already uses all the internal GCSFS functions that already operate on `root_path`.

So for `root_path="some/path"`, `makedir("some/path")` would create `some/path/some/path`